### PR TITLE
gitFeature/pt 4370 get collection meta from sh

### PR DESCRIPTION
**Pull request by dmitry-saraykin**

### CODE REVIEW

File: enrichment/src/test/kotlin/com/rarible/protocol/union/enrichment/meta/collection/provider/SimpleHashCollectionMetaProviderTest.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(MockKExtension::class)
+internal class SimpleHashCollectionMetaProviderTest {
+    @InjectMockKs
+    private lateinit var simpleHashCollectionMetaProvider: SimpleHashCollectionMetaProvider
+
+    @MockK
```
Comment (mystdeim) [inline enrichment/src/test/kotlin/com/rarible/protocol/union/enrichment/meta/collection/provider/SimpleHashCollectionMetaProviderTest.kt:25]: Why do we use different annotations for mocking beans? InjectMocKs and MockK?

**Review by mystdeim — APPROVED** [review]

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/CollectionRepository.kt [hunk 2]
```diff
@@ -103,10 +104,14 @@ class CollectionRepository(
...
 
-    fun findAll(fromIdExcluded: EnrichmentCollectionId? = null): Flow<EnrichmentCollection> = template.find(
+    fun findAll(
+        fromIdExcluded: EnrichmentCollectionId? = null,
+        blockchain: BlockchainDto? = null
+    ): Flow<EnrichmentCollection> = template.find(
         Query(
             Criteria().apply {
                 fromIdExcluded?.let { and(EnrichmentCollection::id).gt(fromIdExcluded) }
+                blockchain?.let { and(EnrichmentCollection::blockchain).isEqualTo(blockchain) }
```
Comment (yakischik) [inline enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/CollectionRepository.kt:114]: Does this query works with existing indexes?

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/service/SimpleHashService.kt [hunk 3]
```diff
@@ -94,8 +104,45 @@ class SimpleHashService(
...
+    suspend fun fetch(key: CollectionIdDto): UnionCollectionMeta? {
+        return fetchRaw(key)?.let { simpleHashConverterService.convert(it) }
+    }
+
+    private suspend fun fetchRaw(key: CollectionIdDto): SimpleHashCollection? {
+        if (!isSupportedCollection(key.blockchain)) {
+            return null
+        }
+        val collection = collectionRepository.get(EnrichmentCollectionId(key))
+        if (collection?.parent != null) {
```
Comment (yakischik) [inline enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/service/SimpleHashService.kt:132]: Better to check structure != REGULAR

File: enrichment/src/test/kotlin/com/rarible/protocol/union/enrichment/meta/collection/provider/SimpleHashCollectionMetaProviderTest.kt (hunk 1 above)
Comment (dmitry-saraykin) [reply enrichment/src/test/kotlin/com/rarible/protocol/union/enrichment/meta/collection/provider/SimpleHashCollectionMetaProviderTest.kt:25]: InjectMocKs is not for mocking. It is to mark where to inject mocked beans.

**Pull request merged**